src/main/java/net/neoforged/neoforge/items/SlotItemHandlerImmutable.java

```diff
@@ -0,0 +1,72 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforge.items;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.Slot;
+import net.minecraft.world.item.ItemStack;
+
+/**
+ * Slot class that can be used with immutable {@link IItemHandler}s
+ * like {@link ComponentItemHandler}
+ */
+public class SlotItemHandlerImmutable extends StackCopySlot {
```

Comment:
I wouldn't use `Immutable` here: the slot is not immutable. `ItemHandlerCopySlot` is a bit better I think (still not great though).